backend/server.js

```diff
@@ -45,10 +44,13 @@ sbp('sbp/selectors/register', {
   'backend/server/handleEntry': async function (entry: GIMessage) {
     const contractID = entry.contractID()
     await sbp('gi.db/log/addEntry', entry)
-    const response = makeResponse(RESPONSE_TYPE.ENTRY, entry.serialize())
     console.log(chalk.blue.bold(`broadcasting ${entry.description()}`))
-    sbp('okTurtles.data/apply', PUBSUB_INSTANCE, p => {
-      p.room(contractID).write(response)
+    sbp('okTurtles.data/apply', PUBSUB_INSTANCE, pubsub => {
+      if (contractID in pubsub.subscribersByContractID) {
+        pubsub.subscribersByContractID[contractID].forEach((client) => {
+          client.send(createMessage('entry', entry.serialize()))
+        })
+      }
```

Comment:
I feel like in most of your code, we have taken several steps forward, without retrace. In this particular section however, it feels to me like a step backward. We went from a single elegant line "to this room, send this message", to complicated logic.

Can you look into reproducing the original API we had with the rooms? Or something similarly elegant so that we can keep this a flexible 1-liner?

Reply:
Agreed this code is a bit low-level, so I am planning to add something like a `broadcast()` method to implement this logic.

Though one single callsite was perhaps not enough to justify the addition of a new method with its own edge cases etc., I have since realized that it was necessary to hide `subscribersByContractID` which should not be accessed outside of the file where it is defined.

Reply:
Sounds good 👍